sale_blanket_order/models/blanket_orders.py

```diff
@@ -0,0 +1,649 @@
+    )
+    invoiced_uom_qty = fields.Float(
+        string="Invoiced quantity", compute="_compute_quantities", store=True
+    )
+    remaining_uom_qty = fields.Float(
+        string="Remaining quantity", compute="_compute_quantities", store=True
+    )
+    remaining_qty = fields.Float(
+        string="Remaining quantity in base UoM",
+        compute="_compute_quantities",
+        store=True,
+    )
+    delivered_uom_qty = fields.Float(
+        string="Delivered quantity", compute="_compute_quantities", store=True
+    )
+    sale_lines = fields.One2many(
+        "sale.order.line",
+        "blanket_order_line",
+        string="Sale order lines",
+        readonly=True,
+        copy=False,
+    )
+    company_id = fields.Many2one(
+        "res.company", related="order_id.company_id", store=True
+    )
+    currency_id = fields.Many2one("res.currency", related="order_id.currency_id")
+    partner_id = fields.Many2one(related="order_id.partner_id", string="Customer")
+    user_id = fields.Many2one(related="order_id.user_id", string="Responsible")
+    payment_term_id = fields.Many2one(
+        related="order_id.payment_term_id", string="Payment Terms"
+    )
+    pricelist_id = fields.Many2one(related="order_id.pricelist_id", string="Pricelist")
+
+    price_subtotal = fields.Monetary(
+        compute="_compute_amount", string="Subtotal", store=True
+    )
+    price_total = fields.Monetary(compute="_compute_amount", string="Total", store=True)
+    price_tax = fields.Float(compute="_compute_amount", string="Tax", store=True)
+
+    def name_get(self):
+        result = []
+        if self.env.context.get("from_sale_order"):
+            for record in self:
+                res = "[%s]" % record.order_id.name
+                if record.date_schedule:
+                    formatted_date = format_date(record.env, record.date_schedule)
+                    res += " - {}: {}".format(_("Date Scheduled"), formatted_date)
+                res += " ({}: {} {})".format(
+                    _("remaining"), record.remaining_uom_qty, record.product_uom.name,
+                )
+                result.append((record.id, res))
+            return result
+        return super().name_get()
+
+    def _get_real_price_currency(self, product, rule_id, qty, uom, pricelist_id):
+        """Retrieve the price before applying the pricelist
+            :param obj product: object of current product record
+            :param float qty: total quentity of product
+            :param tuple price_and_rule: tuple(price, suitable_rule) coming
+                   from pricelist computation
+            :param obj uom: unit of measure of current order line
+            :param integer pricelist_id: pricelist id of sale order"""
+        # Copied and adapted from the sale module
+        PricelistItem = self.env["product.pricelist.item"]
+        field_name = "lst_price"
+        currency_id = None
+        product_currency = None
```

Comment:
You can assign an empty model instance as default instead of using None. This way, you don't have to worry about if you can get `.id` or any other instance attribute.

```suggestion
        currency_id = self.env["res.currency"]
        product_currency = self.env["res.currency"]
```